src/createSwappableProxy.ts

```diff
@@ -0,0 +1,82 @@
+import type { SwappableProxy } from './types';
+
+/**
+ * Creates a proxy object that initially points to the given object but whose
+ * target can be substituted with another object later using its `setTarget`
+ * method.
+ *
+ * @param initialTarget - The initial object you want to wrap.
+ * @returns The proxy object.
+ */
+export function createSwappableProxy<T extends object>(
+  initialTarget: T,
+): SwappableProxy<T> {
+  let target = initialTarget;
+
+  /**
+   * Changes the object that the proxy wraps.
+   *
+   * @param newTarget - The new object.
+   */
+  let setTarget = (newTarget: T) => {
+    target = newTarget;
+  };
+
+  const proxy = new Proxy<T>(target, {
+    // @ts-expect-error We are providing a different signature than the `get`
+    // option as defined in the Proxy interface; specifically, we've limited
+    // `name` so that it can't be arbitrary. Theoretically this is inaccurate,
+    // but because we've constrained what the `target` can be, that effectively
+    // constraints the allowed properties as well.
+    get(
+      _target: any,
```

Comment:
Nit: it doesn't really matter, but we can set `_target` to `T` here. That's what the getter type signature is expecting anyway. The same goes for `set` as well.

Reply:
Yeah, makes sense. Changed in [da7c9fb](https://github.com/MetaMask/swappable-obj-proxy/pull/27/commits/da7c9fb9f5b0f49ac7f02aae7d42af2f9c93dcea).

Reply:
Thanks! Though like you adjusted the type for `get` in one file and `set` in the other, but both can be set to T in both files

Reply:
Oops, good catch. All should be updated as of [ccc75d5](https://github.com/MetaMask/swappable-obj-proxy/pull/27/commits/ccc75d532a1430c9d880fb7648195172e2e0fa0a).